### PR TITLE
Helicopter Gunship uniques

### PR DESCRIPTION
**Pull request by givehub99**
Preparation for Helicopter Gunship (vanilla) unit and Carthaginian unique (G&K)

-"All tiles costs 1" unique makes every tile cost 1. Used by Helicopter in CIV 5.

-"Can pass through impassable tiles" unique allows unit to pass through any tile. Used by Helicopter in CIV 5. Does not let units cross ocean if they couldn't already, just terrain that is considered "impassable"

-Mountains now do 50 damage to units that end turn on it. In civ 5 and its expansion this is what happens to any unit that is able to pass mountains.
This is "hard coded" for now. Also added the notification text to template.properties

### CODE REVIEW

File: core/src/com/unciv/logic/map/MapUnit.kt [hunk 1]
```diff
@@ -657,6 +662,20 @@ class MapUnit {
                 .sumBy { it.params[0].toInt() }
     }
 
+    private fun getTerrainDamage() {
+        // hard coded mountain damage for now
+        if (getTile().baseTerrain == Constants.mountain) {
+            val tileDamage = 50
```
Comment (yairm210) [inline core/src/com/unciv/logic/map/MapUnit.kt:668]: takeDamage(50)

File: core/src/com/unciv/logic/map/TileMap.kt [hunk 2]
```diff
@@ -184,7 +184,9 @@ class TileMap {
 
     fun getViewableTiles(position: Vector2, sightDistance: Int): List<TileInfo> {
         val viewableTiles = getTilesInDistance(position, 1).toMutableList()
-        val currentTileHeight = get(position).getHeight()
+        val tile = get(position)
+        // mountains do not give sight bonus when on them
+        val currentTileHeight =  if (tile.baseTerrain == Constants.mountain) 0 else tile.getHeight()
```
Comment (yairm210) [inline core/src/com/unciv/logic/map/TileMap.kt:189]: So, if you're on a mountain you can't see over hills?

File: core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt [hunk 3]
```diff
@@ -8,6 +8,8 @@ class UnitMovementAlgorithms(val unit:MapUnit) {
 
     // This function is called ALL THE TIME and should be as time-optimal as possible!
     fun getMovementCostBetweenAdjacentTiles(from: TileInfo, to: TileInfo, civInfo: CivilizationInfo): Float {
+        if (unit.allTilesCosts1)
```
Comment (yairm210) [inline core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt:11]: I'm pretty sure that helicopters also get the +1 movement penalty from the great wall

File: core/src/com/unciv/logic/map/MapUnit.kt (hunk 1 above)
Comment (givehub99) [reply core/src/com/unciv/logic/map/MapUnit.kt:668]: I did it the same way citadel damage is done from MapUnit.kt:

```
    private fun getCitadelDamage() {
        // Check for Citadel damage
        val applyCitadelDamage = currentTile.neighbors
                .filter { it.getOwner() != null && civInfo.isAtWarWith(it.getOwner()!!) }
                .map { it.getTileImprovement() }
                .filter { it != null && it.hasUnique("Deal 30 damage to adjacent enemy units") }
                .any()

        if (applyCitadelDamage) {
            health -= 30

            if (health <= 0) {
                civInfo.addNotification("An enemy [Citadel] has destroyed our [$name]", currentTile.position, Color.RED)
                // todo - add notification for attacking civ
                destroy()
            } else civInfo.addNotification("An enemy [Citadel] has attacked our [$name]", currentTile.position, Color.RED)
        }
    } 
```

As far as I'm aware, takeDamage isn't available to the MapUnit class. Should terrain damage be put somewhere else?

File: core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt (hunk 3 above)
Comment (givehub99) [reply core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt:11]: -Reverted mountains not giving sight bonus. I think I was misled by the comment in the units json file saying that mountains don't give a vision bonus. Seems that was incorrect? I checked in Civ 5 and helicopters on mountains can see over hills.
-Helicopters do not get a movement penalty from great wall. Tested in Civ 5 using map editor and loading the scenario - I tested with a knight which did get a movement penalty and I tested with a helicopter which didn't get a movement penalty.

File: core/src/com/unciv/logic/map/MapUnit.kt (hunk 1 above)
Comment (yairm210) [reply core/src/com/unciv/logic/map/MapUnit.kt:668]: There isn't?!
Huh
Maybe it's in the combattant class
K then

**Pull request merged**